pkg/util/maven/maven_command.go

```diff
@@ -142,7 +142,12 @@ func (c *Command) Do(ctx context.Context) error {
 
 	Log.WithValues("MAVEN_OPTS", mavenOptions).Infof("executing: %s", strings.Join(cmd.Args, " "))
 
-	return util.RunAndLog(ctx, cmd, mavenLogHandler, mavenLogHandler)
+	// generate maven file
+	if err := generateMavenContext(c.context.Path, args); err != nil {
```

Comment:
I am wondering if, instead of copying the execution config in a file, you could try to regenerate the same at the moment of calling the JIB (ie, creating a func which take care to generate and call it here and later when needed). It would be much cleaner IMO.

Reply:
It would be better, but I would like to dedicate another issue to this part as it needs a big refactoring in a sensitive part of the build task. It would be also an occasion to see if we can separate the maven code dedicate to commands execution from the project generation code.